src/core/controller.ts

```diff
@@ -8,6 +8,14 @@ import { ValuePropertiesBlessing, ValueDefinitionMap } from "./value_properties"
 
 export type ControllerConstructor = Constructor<Controller>
 
+type DispatchOptions = {
+  target?: Element | Window | Document
+  detail?: Object
+  prefix?: string
+  bubbles?: boolean
+  cancelable?: boolean
```

Comment:
Instead of marking each property as optional, what do you think about changing `dispatch` to accept a `Partial<DispatchOptions>`?

Reply:
Even better, thanks!